librz/core/golang.c

```diff
@@ -1605,14 +1612,80 @@ static ut32 golang_recover_string_riscv64(GoStrRecover *ctx) {
 	return 4;
 }
 
+// Sometimes the data-structures has strings, but these are stored in tables where
+// the first offset is always the pointer to the squashed strings and the next word
+// is the size of the string
+static void core_recover_golang_strings_from_data_pointers(RzCore *core, GoStrRecover *ctx) {
+	rz_core_notify_begin(core, "Recovering go strings from the data sections");
+
+	RzAnalysis *analysis = core->analysis;
+	const ut32 word_size = analysis->bits / 8;
+	RzListIter *iter;
+	RzBinSection *section;
+	ut8 *buffer = NULL;
+	ut64 string_addr, string_size;
+	RzList *section_list = rz_bin_get_sections(core->bin);
```

Comment:
Do we really need to look at sections? We should tend to avoid relying on sections (and  section names ) if possible.

Reply:
actually yes, i could just check if it does not have exec flag, but some PE has data that is executable

Reply:
really?? that is weird, especially for common windows binaries

Reply:
I'm still not sure about this... By using `rz_bin_get_sections` we are just getting the sections of the binary which, as said elsewhere, do not really mean too much for ELF, for example. For ELF we should use segments, but for other binaries maybe sections are enough... I think in general we may just want to look at `RzBinMap`, which represent a memory mapping from a binary into the virtual memory.

Another reason for using RzBinMap is that actually we want to look at the mapped portions of a binary and sections may contain stuff like .dynamic, .comment, etc. which are not mapped anywhere (thus the read would fail).

Reply:
i agree but in this case, we do not care, because these structures are pointing to RO memory normally on golang bins. since we do have different bins using golang we only check for "data" sections/segments/etc..

Reply:
I think our users would like to reverse even binaries that are stripped and do not have non essential information (e.g. sections). Rizin should try to do its best with a regularly compiled binary but also with a manipulated one.

Do you see any drawback for the regularly compiled binaries in using the rzbinmap approach instead of the sections?

Reply:
no.